src/rwa/components/table/formatted-number.tsx

```diff
@@ -0,0 +1,32 @@
+import { ComponentPropsWithoutRef } from 'react';
+import { NumericFormat } from 'react-number-format';
+
+type Props = Omit<
+    ComponentPropsWithoutRef<typeof NumericFormat>,
+    'displayType'
+> & {
+    currency?: 'USD';
+};
+
+export function FormattedNumber(props: Props) {
```

Comment:
wrapper to make it easy to use the default props as described by teep. `displayType="text"` just renders a span.